src/printer.js

```diff
@@ -4256,7 +4262,19 @@ function printAstToDoc(ast, options) {
 
   const doc = printGenerically(FastPath.from(ast));
   docUtils.propagateBreaks(doc);
```

Comment:
This should likely be done after you add hardline as hardline intro a break. This isn't going to do anything wrong here but it would be better to do the "right" thing as if we're adding some more code that relies on this, it's going to break in unexpected ways.